### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Adding a test infrastructure while doing some minimal cleanup and fixing things.

### CODE REVIEW

**Comment by jaimergp** [comment]
Ready for review @conda/constructor :D

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,7 @@
+* text=auto
+
+*.patch binary
```
Comment (jaimergp) [inline .gitattributes:3]: Without this line, patches can't be applied due to line endings 🤯

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          environment-file: false
+          channels: conda-forge
+          extra-specs: |
+            python=${{ matrix.python-version }}
+            conda-build
+            conda-libmamba-solver
+      - name: Build recipe
+        shell: bash -el {0}
+        env:
+          CONDA_SOLVER: libmamba
```
Comment (jaimergp) [inline .github/workflows/tests.yml:45]: You can use the new solver with `conda-build` just by setting this line. It should be equivalent to `mambabuild`.

File: src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
++            fh.write(
++                f'SET "PATH={prefix};'
++                f'{prefix}\\Library\\mingw-w64\\bin;'
++                f'{prefix}\\Library\\usr\\bin;'
++                f'{prefix}\\Library\\bin;'
++                f'{prefix}\\Scripts;'
++                f'{prefix}\\bin;'
++                '%PATH%"'
++            )
+             fh.write(f"{silencer}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n")
```
Comment (jezdez) [inline src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch:25]: I'd really love to understand this patch!

**Review by jezdez — COMMENTED** [review]
Looks good to me on a high-level, the $PATH manipulation freaks me out a little.

**Comment by jezdez** [comment]
@jaimergp I've also added a CODEOWNERS file to ease of reviews: https://github.com/conda/conda-standalone/pull/4/commits/398825f03538b0168873ccfc075f769ccb0edc48

File: src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch (hunk 3 above)
Comment (jaimergp) [reply src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch:25]: Same :D 

Earliest [history bit](https://github.com/AnacondaRecipes/conda-standalone-feedstock/commit/ae5b5ef6411531c71eac776881951d1451b7f082) I could find already included this file without explanation, so I am guessing it comes from a different repot (the patch date predates the repo creation date).

@nehaljwani - do you recall any details?

Comment (nehaljwani) [reply src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch:25]: IIRC, there is no `conda.bat` file to call activate on, when it is invoked in the frozen state.

**Comment by jaimergp** [comment]
@jezdez Is this good to go for now?

**Review by jezdez — APPROVED** [review]

File: src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch (hunk 3 above)
Comment (isuruf) [reply src/conda_patches/0002-Manipulate-PATH-directly-instead-of-_call_ing-conda.patch:25]: There's no corresponding unix patch though.

**Pull request merged**